ammico/notebooks/DemoNotebook_ammico_MultimodalSearch.ipynb

```diff
@@ -0,0 +1,485 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Here goes the code that reads in multiple images as reference\n",
+    "# then you will loop over these multiple images and find the best matches\n",
+    "# in the end, the best matches will be averaged over for each picture and a list of averaged best matches will be provided"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import importlib_resources                                                                       # only require for image query example\n",
+    "# image_example_query = str(importlib_resources.files(\"ammico\") / \"data\" / \"test-crop-image.png\")   # creating the path to the image for the image query example\n",
+    "image_example_query = \"../../../test-images-politicians/5e4c7722-d620-42ab-915e-9b1a048e2992Original.jpg\"  # creating the path to the image for the image query example\n",
+    "image_example_query2 = \"../../../test-images-politicians/service-pnp-ppbd-00600-00608_150px.jpg\"\n",
+    "\n",
+    "search_query = [\n",
+    "    {\"image\": image_example_query},      # This is how looks image query, here `image_example_path` is the path to query image like \"data/test-crop-image.png\"\n",
+    "    {\"image\": image_example_query2},\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "similarity, sorted_lists = my_obj.multimodal_search(\n",
+    "    model,\n",
+    "    vis_processors,\n",
+    "    txt_processors,\n",
+    "    model_type,\n",
+    "    image_keys,\n",
+    "    features_image_stacked,\n",
+    "    search_query,\n",
+    "    filter_number_of_images=20,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "similarity # now a 2D tensor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# average similarities\n",
+    "print(similarity.mean(dim = 1)) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# add the similarity average to the image_dict\n",
+    "for key in image_dict.keys():\n",
+    "    # find the similarities for each image in the search query\n",
+    "    similarities = [image_dict[key][query[\"image\"]] for query in search_query]\n",
+    "    image_dict[key][\"similarity_average\"] = sum(similarities)/len(similarities)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Not yet compatible with show_results due to dictionary design"
```

Comment:
Can you add two to three cells below, where the dict is converted to a df, and the df is shown and then written to csv? Using `df = ammico.get_dataframe(image_dict)`, I believe in the original demo_notebook there is an outdated line in the Multimodal search from before when we created the wrapper for the output.